book-content/chapters/07-mutability.md

```diff
@@ -0,0 +1,1013 @@
+
+But there's another interesting solution.
+
+#### How TypeScript Infers `const`
+
+When using `const`, the variable is _immutable_ and cannot be reassigned. When we change the variable declaration to use `const`, TypeScript will infer the type more narrowly:
+
+```ts twoslash
+// @errors: 2345
+type AlbumGenre = "rock" | "country" | "electronic";
+
+// ---cut---
+const albumGenre = "rock";
+
+const handleGenre = (genre: AlbumGenre) => {
+  // ...
+};
+
+handleGenre(albumGenre); // No error
+```
+
+There is no longer an error in the assignment, and hovering over `albumGenre` inside of the `albumDetails` object shows that TypeScript has inferred it as the literal type `"rock"`.
+
+If we try to change the value of `albumGenre` after declaring it as `const`, TypeScript will show an error:
+
+```ts twoslash
+// @errors: 2588
+type AlbumGenre = "rock" | "country" | "electronic";
+
+const albumGenre = "rock";
+
+// ---cut---
+albumGenre = "country";
+```
+
+TypeScript is mirroring JavaScript's treatment of const in order to prevent possible runtime errors. When you declare a variable with `const`, TypeScript infers it as the literal type you specified.
+
+So, TypeScript uses how JavaScript works to its advantage. This will often encourage you to use `const` over `let` when declaring variables, as it's a little stricter.
+
+### Object Property Inference
+
+The picture with `const` and `let` becomes a bit more complicated when it comes to object properties.
+
+Objects are mutable in JavaScript, meaning their properties can be changed after they are created.
+
+For this example, we have an `AlbumAttributes` type that includes a `status` property with a union of literal values representing possible album statuses:
+
+```typescript
+type AlbumAttributes = {
+  status: "new-release" | "on-sale" | "staff-pick";
+};
+```
+
+Say we had an `updateStatus` function that takes an `AlbumAttributes` object:
+
+```ts twoslash
+// @errors: 2345
+type AlbumAttributes = {
+  status: "new-release" | "on-sale" | "staff-pick";
+};
+// ---cut---
+const updateStatus = (attributes: AlbumAttributes) => {
+  // ...
+};
+
+const albumAttributes = {
+  status: "on-sale",
+};
+
+updateStatus(albumAttributes);
+```
+
+TypeScript gives us an error below `albumAttributes` inside of the `updateStatus` function call, with messages similar to what we saw before.
+
+This is happening because TypeScript has inferred the `status` property as a `string` rather than the specific literal type `"on-sale"`. Similar to with `let`, TypeScript understands that the property could later be reassigned:
+
+```typescript
+albumAttributes.status = "new-release";
+```
+
+This is true even though the `albumAttributes` object was declared with `const`. We get the error when calling `updateStatus` because `status: string` can't be passed to a function that expects `status: "new-release" | "on-sale" | "staff-pick"`. TypeScript is trying to protect us from potential runtime errors.
```

Comment:
> This is true even though the `albumAttributes` object was declared with `const`

I feel like this should be explained explicitly here - less experienced readers might get lost here as it seemingly contradicts some of the things said earlier. I know that this section also mentions smth like:
> Objects are mutable in JavaScript, meaning their properties can be changed after they are created.

but I still feel like it might be a little bit hard to connect the dots here